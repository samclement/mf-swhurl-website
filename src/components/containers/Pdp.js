import React from 'react'
import Head from 'next/head'
import Header from '../blocks/Header'
import { getProduct } from '../../services'
import GenderHeader from '../elements/GenderHeader.js'
import styled from 'styled-components'
import ReactSwipe from 'react-swipe'

const StyledProductContainer = styled.div`
  width: 100%;
  @media only screen and (min-width: 641px) {
    display: inline-block;
    text-align: center;
  }
`

const ImageContainer = styled.div`
  width: 100%;
  @media only screen and (min-width: 641px) {
    width: 44%;
    display: inline-block;
    padding: 10px 3% 0 3%;
  }
  float: left;
`

const StyledImage = styled.img`
  width: 100%;
  margin: 0 auto;
  @media only screen and (max-width: 320px) {
    max-width: 320px;
  }
`

const ProductDetailsContainer = styled.div`
  @media only screen and (min-width: 641px) {
    width: 50%;
    display: inline-block;
  }
`

const Controls = styled.div`
  text-align: center;
  margin-bottom: 20px;
`

function Pdp(props) {
  const { name, thumbnail, designerData, priceData, sizeTaxonomyName } = props
  const mainImage = thumbnail.replace('thumbnail', 'large')
  const gender = sizeTaxonomyName.split(' ')[0].toLowerCase()
  let reactSwipeEl
  return (
    <div>
      <Head>
        <title>mf.swhurl.com | {name}</title>
      </Head>
      <Header gender={gender} />
      <GenderHeader gender={gender} />
      <StyledProductContainer>
        <ImageContainer>
          <ReactSwipe className="carousel" ref={el => (reactSwipeEl = el)}>
            {[...Array(5).keys()].map(num => {
              return (
                <div key={num}>
                  <StyledImage
                    src={mainImage.replace('_1', `_${num + 1}`)}
                    alt={name}
                  />
                </div>
              )
            })}
          </ReactSwipe>
          <noscript>
            <img src={mainImage} />
          </noscript>
          <Controls>
            <button onClick={() => reactSwipeEl.prev()}>Previous</button>
            <button onClick={() => reactSwipeEl.next()}>Next</button>
          </Controls>
        </ImageContainer>
        <ProductDetailsContainer>
          <h2 dangerouslySetInnerHTML={{ __html: designerData.name }} />
          <h3 dangerouslySetInnerHTML={{ __html: name }} />
          <p dangerouslySetInnerHTML={{ __html: priceData.formattedValue }} />
        </ProductDetailsContainer>
      </StyledProductContainer>
    </div>
  )
}

Pdp.getInitialProps = async ({ req, query }) => {
  const res = await getProduct(query, !!req)
  return res.data
}

export default Pdp
