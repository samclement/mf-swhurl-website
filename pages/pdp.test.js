/* eslint-env jest */

import React from 'react'
import { render } from 'react-testing-library'
import 'jest-styled-components'
import 'jest-dom/extend-expect'
import Pdp from './pdp.js'
// Nextjs routing support
import Router from 'next/router'
const mockedRouter = { push: () => {}, prefetch: () => {} }
Router.router = mockedRouter
// Data
const data = require('../test-data/pdp.json')

describe('With Snapshot Testing', () => {
  it('Plp page shows "mf.swhurl.com" H1', () => {
    const { asFragment } = render(<Pdp {...data} />)
    expect(asFragment()).toMatchSnapshot()
  })
  it('Plp getInitialProps', async () => {
    const res = await Pdp.getInitialProps({ query: { code: 1247661 } })
    expect(res).toMatchSnapshot()
  })
  it('Plp getInitialProps', async () => {
    const res = await Pdp.getInitialProps({ req: { query: { code: 1247661 } } })
    expect(res).toMatchSnapshot()
  })
})
