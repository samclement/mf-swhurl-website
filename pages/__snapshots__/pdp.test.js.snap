// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Plp getInitialProps 1`] = `Object {}`;

exports[`With Snapshot Testing Plp getInitialProps 2`] = `Object {}`;

exports[`With Snapshot Testing Plp page shows "mf.swhurl.com" H1 1`] = `
.c2 {
  color: black;
  padding-right: 5px;
}

.c2:hover {
  color: #F00;
}

.c2:visited {
  color: black;
}

.c1 {
  font-weight: bold;
  text-transform: capitalize;
  padding-right: 10px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
}

.c5 {
  width: 100%;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c6 {
  min-width: 320px;
}

@media only screen and (min-width:640px) {
  .c4 {
    max-width: 320px;
  }
}

@media only screen and (min-width:800px) {
  .c4 {
    max-width: 500px;
  }
}

@media only screen and (max-width:320px) {
  .c5 {
    max-width: 320px;
  }
}

@media only screen and (min-width:640px) {
  .c5 {
    margin-right: 50px;
  }
}

@media only screen and (min-width:640px) {
  .c3 {
    -webkit-flex-wrap: nowrap;
    -ms-flex-wrap: nowrap;
    flex-wrap: nowrap;
  }
}

<div>
  <nav
    class="c0"
  >
    <div
      class="c1"
    >
      mens
    </div>
    <a
      class="c2"
      href="/mens/shop/accessories"
    >
      Accessories
    </a>
    <a
      class="c2"
      href="/mens/shop/shoes"
    >
      Shoes
    </a>
    <a
      class="c2"
      href="/mens/shop/clothing"
    >
      Clothing
    </a>
    <a
      class="c2"
      href="/mens/shop/bags"
    >
      Bags
    </a>
    <a
      class="c2"
      href="/mens/shop/homeware"
    >
      Homeware
    </a>
  </nav>
  <nav
    class="c0"
  >
    <div
      class="c1"
    >
      womens
    </div>
    <a
      class="c2"
      href="/womens/shop/accessories"
    >
      Accessories
    </a>
    <a
      class="c2"
      href="/womens/shop/bags"
    >
      Bags
    </a>
    <a
      class="c2"
      href="/womens/shop/clothing"
    >
      Clothing
    </a>
    <a
      class="c2"
      href="/womens/shop/shoes"
    >
      Shoes
    </a>
    <a
      class="c2"
      href="/womens/shop/homeware"
    >
      Homeware
    </a>
  </nav>
  <div
    class="c3"
  >
    <div
      class="c4"
    >
      <img
        alt="Intrecciato leather cardholder"
        class="c5"
        src="//assetsprx.matchesfashion.com/img/product/1247661_1_large.jpg"
      />
    </div>
    <div
      class="c6"
    >
      <h1>
        Bottega Veneta
      </h1>
      <h2>
        Intrecciato leather cardholder
      </h2>
      <p>
        Â£265
      </p>
    </div>
  </div>
</div>
`;
